odoo/addons/base/models/ir_binary.py

```diff
@@ -45,14 +45,7 @@ def _find_record(
         if not record:
             raise MissingError(f"No record found for xmlid={xmlid}, res_model={res_model}, id={res_id}")
 
-        record = self._find_record_check_access(record, access_token, field)
-        return record
-
-    def _find_record_check_access(self, record, access_token, field):
-        if record._name == 'ir.attachment':
-            return record.validate_access(access_token)
-
-        record.check_access('read')
+        record = record._find_record_check_access(access_token, field)
```

Comment:
As discussed, I think check access and sudo (when necessary) should just be moved here.

The method on models should just return whether access can be bypassed for a specific case, as this is why we use it exactly. It doesn't "find a record" and it more often than not also doesn't "check access".